### ISSUE TITLE
Debate: Fix for Background "Seams"

### ISSUE THREAD

**Issue by lemoness** [HabitRPG/habitica#3758]
We've gotten a fair amount of complaints about background "seams", which is to say, backgrounds that don't mesh. There have been three main suggestions that are not completely infeasible (i.e. "have three versions of each background so that they can create a cohesive scene").
1. Have all future backgrounds tile, so that at least people have an option to match them.
2. (related to #1) Have an over-ride selection, so that you can display the tiled version of your background behind all of your party-mates.
3. Have "frames" or separators between the backgrounds.
#2 is feature requiring programming, so we are going to set it aside for now, and examine #1 and #3 for the time being.

Here is are some examples of the current way backgrounds function.

![backgroundcomp1b](https://cloud.githubusercontent.com/assets/3494081/3635767/d82e0ec6-0f89-11e4-8ba8-6e993968d4d9.png)
![backgroundcomp2b](https://cloud.githubusercontent.com/assets/3494081/3635765/d8209e08-0f89-11e4-84b8-a4bfe7c5186b.png)
![backgroundcomp3a](https://cloud.githubusercontent.com/assets/3494081/3635766/d821e272-0f89-11e4-826c-d2e8b62e3f81.png)
![backgroundcomp2a](https://cloud.githubusercontent.com/assets/3494081/3635764/d8200524-0f89-11e4-8708-06f5a106fd81.png)

Pros: Maximum creativity for artists
Cons: Seam between same backgrounds as well as different backgrounds, leading to an odd visual effect

Here is an example of two "tiled" backgrounds

![backgroundcomp1a](https://cloud.githubusercontent.com/assets/3494081/3635785/40893590-0f8a-11e4-80e9-a68503b0fbe4.png)

Pros: Allows for a more unified look, especially if there is an "override" option coded in later. Looks most polished.
Cons: Does not eliminate seams between different backgrounds + background + no background. Also, some artists have expressed concern that it limits their creativity.

Here are some divider lines at 1 px. They could be a separate layer, or incorporated as part of the background art.
![backgroundcomp4a](https://cloud.githubusercontent.com/assets/3494081/3635807/a745a098-0f8a-11e4-91df-e18dc147c616.png)
![backgroundcomp4b](https://cloud.githubusercontent.com/assets/3494081/3635809/a756dc6e-0f8a-11e4-90f5-8062287244d3.png)
![backgroundcomp4c](https://cloud.githubusercontent.com/assets/3494081/3635808/a7483e34-0f8a-11e4-84db-491c6858c358.png)
![backgroundcomp4d](https://cloud.githubusercontent.com/assets/3494081/3635810/a769b7b2-0f8a-11e4-9225-387d6aa2c7a5.png)

Pros: Separates backgrounds clearly in all scenarios.
Cons: Not as visually appealing as tiling; 1 px division looks strange with 3 px art; covers some of the art. For backgrounds that DO tile, the pleasant effect is lost, which is particularly problematic if people want the override feature, which has been requested several times.  (Note that in this version, the line between two backgrounds is 2 px wide. A different version could be made that would have only 1 px wide by making the diving line appear only on the left-most side of the background, but this might look strange for the user's avatar background.)

Here is a 3px-wide separator line.

![backgroundcomp5a](https://cloud.githubusercontent.com/assets/3494081/3635828/4b479f0c-0f8b-11e4-86d3-fa14a4ca3c2f.png)
![backgroundcomp5b](https://cloud.githubusercontent.com/assets/3494081/3635827/4b39b6a8-0f8b-11e4-99bf-04a9279c533a.png)
![backgroundcomp5c](https://cloud.githubusercontent.com/assets/3494081/3635826/4b392ada-0f8b-11e4-80f8-ea27fe43c967.png)

Pros: fits the 3 px size.
Cons: All the cons of the 1 px separators, plus the additional disadvantage of obscuring more of the background art and looking more physically "jarring."

Some other options: changing the color of frames (to the blue-gray default background color, or a color that matches the backgrounds), experimenting with thickness, etc.

Feel free to weigh in on opinions about these options, as well as to share new ideas!

**Comment by deilann** [comment]
I was thinking the backgrounds themselves would have the dividers build
into them as a frame, at 3px in a color that matched them. No need for an
extra layer and no worries about it looking weird because it will certainly
be the right color as it's built into the background.

And if that's the case, it could be that you either have to build in a
frame or you have to have it tiling.

**Comment by breadstrings** [comment]
The con on having the borders built in (with different colours) is that it could look funky when placed next to each other. Also, it'd require a 1.5 pixel border around each background in order to get 3 pixels between each background (not feasible). Otherwise, the pixel borders would either be two pixels wide or 4 pixels wide. 

I'm a fan of the override function but as Lemoness said, it'd be hard to code.

I can't remember who suggested the idea of actual decorative frames acting like a 'window' into your party's avatars, but what if we did that? Go full into the concept of "window" and make the frames an intentional feature. Like maybe  a circle frame or a guassian blur around each avatar (more like a dream sequence type of thing) with grey being the dividing colour.

**Comment by deilann** [comment]
It really just depends on how the borders are done. I mean, it doesn't
necessarily need to be a "single width pixel border" right? It just needs
to keep it from looking like "this should almost tile, but doesn't"

**Comment by breadstrings** [comment]
True, but we also need a border that is uniform and can sit next to each other easily. What if we simply separated the backgrounds with the a grey border the same colour as the non-background avatars?

**Comment by deilann** [comment]
I disagree that we need a border that's uniform and can sit next to each
other easily. The "seams" between the non-similar backgrounds don't feel
like as much a problem as the seams between the ones that almost look like
they should tile and don't. I think the solution would be to not have
"almost tile but doesn't," which could be done sometimes with a border and
sometimes by switching up the art.

**Comment by k8sullyvan** [comment]
I personally like the 1px border the most if we go down the border route; or borders only on the backgrounds that won't tile with themselves. I think that when there are 2 backgrounds next to each other, that are different, then it doesn't make a massive amount of difference between whether there is a line between them or not.

**Comment by Pandah** [comment]
I also don't think two separate color borders built into the image will look odd next to one another. I did a little bit of a mockup of what I imagined for them.

![border1](https://cloud.githubusercontent.com/assets/6396120/3637983/5118d984-102e-11e4-9e13-9a857f0b6d95.png)
![border2](https://cloud.githubusercontent.com/assets/6396120/3637984/55e02d82-102e-11e4-8363-170e807724dd.png)
![border3](https://cloud.githubusercontent.com/assets/6396120/3637985/58bf1220-102e-11e4-8e94-68911013e356.png)

Because it would be up to the artist to add in the border, it could be ensured that it would be subtle or at least matching so that it doesn't look as jarring as the black outline.

Oh, and here is what it would look like if you did the either-or suggestion I gave at the PIP guild.
![border4](https://cloud.githubusercontent.com/assets/6396120/3637994/ae5f221a-102e-11e4-84e4-a64a8d30f759.png)
Basically, ensure that all backgrounds either have a border built in or tile. If two different tiling backgrounds are placed next to one another they will look like the third example on Lemoness' original post.

**Comment by lemoness** [comment]
Thanks for the mock-ups, Pandah!

I do like the look of the dark colored border for both, even the tiling
ones. I think that having a border and an unborded background next to each
other, (especially if the next party member then has another unborded,
different tiling background) will start to look sloppier and sloppier the
more people you have in your party. I think that the tiling effect is not
actually diminished as muh as I worried it would be when you are using dark
colored borders that correspond to the background colors, so my vote is
definitely for these borders on all of them.

**Comment by Pandah** [comment]
I think it helped that I lowered the opacity on the border I made so that you could still see the artwork behind it. It is an easy technique (make a new layer for the border, slide the opacity bar to your desired visual, for those that may not know. these are all about 64%. Or you can set the layer to multiply to get a similar effect.) and will still allow the tiled backgrounds to match up underneath the border, but give enough division between different backgrounds. This probably isn't a necessary technique, but would definitely make it so tiling backgrounds are still an option for artists who want to do so. :)

I think my vote is for artist-added borders for all, too.

**Comment by SabreCat** [comment]
I love the way the tiled backgrounds look, but @Pandah's transparent borders are also quite nice!

**Comment by lemoness** [comment]
Another vote for the semi-transparent borders from starsystemic (Olivia Schwartz) in the Pixels in Progress guild!

**Comment by deilann** [comment]
I'll add my support behind the semi-transparent borders. I think they look
lovely.

**Comment by Draayder** [comment]
I also vote for the semi-transparent borders, they don't look jarring but keep a good degree of separation between backgrounds.

**Comment by Gaelan** [comment]
What would be _awesome_ but a LOT of work for pixelists would be to have a 10-or-so px "transition" area for each background combination.

**Comment by deilann** [comment]
That would become exponential work, and quickly unfeasible.

**Comment by Gaelan** [comment]
Yeah, sadly.

**Comment by lemoness** [comment]
Okay, we will be going with Pandah's solution! Thank you all for your helpful feedback :)
